Pytest 8.0.2 may collect tests multiple times under Windows

I'm not quite sure if this is a bug, or a problem with the test or GitHub Actions, but since version 8.0.2 a test in `pytest-order` fails in the CI under Windows, which worked fine in 8.0.1 and previous versions. Linux and macOS tests are not affected.

Pytest run environment in the CI:
```
platform win32 -- Python 3.11.8, pytest-8.0.2, pluggy-1.4.0 -- D:\a\pytest-order\pytest-order\.tox\py311-pytest80\Scripts\python.EXE
cachedir: .tox\py311-pytest80\.pytest_cache
rootdir: D:\a\pytest-order\pytest-order
plugins: cov-2.9.0, dependency-0.6.0, mock-3.12.0, xdist-3.5.0
```
The same happens for Python 3.8 - 3.12.

The [actual test](https://github.com/pytest-dev/pytest-order/blob/main/tests/test_xdist_handling.py) should collect 6 items (3 tests per test module, in 2 modules) that are created in the directory provided by the `tempdir` fixture, 
but collects 24 tests in the CI instead. Locally, it works as expected.

Checking the failed tests revealed that the tests have been collected from different locations. The tempdir had been at:
`C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-9\\test_xdist_ordering0` 
(for a specific run), but the same tests have been discovered with a test ID with just the qualified test name, and additionally with the test IDs:
`::::Users::runneradmin::AppData::Local::Temp::pytest-of-runneradmin::pytest-9:::xxx` and 
`::::Documents and Settings::runneradmin::AppData::Local::Temp::pytest-of-runneradmin::pytest-9::xxx`,
which are most likely aliases.

I haven't dug into this yet, but I suspect it may have something to do with the fix for [the problem with short paths](https://github.com/pytest-dev/pytest/issues/11895).

Actually I wrote a test very similar to yours, and it passed locally, but failed in the CI (though with 2 tests passed, not 8, in this case):
```
    @pytest.mark.skipif(not sys.platform.startswith("win"), reason="Windows only")
    def test_collect_symlinks_windows(pytester: Pytester, tmpdir) -> None:
        """Regression test for #12039: Tests collected multiple times under Windows."""
        test_file = Path(tmpdir) / "symlink_collection_test.py"
        test_file.write_text("def test(): pass", encoding="UTF-8")
        result = pytester.runpytest(tmpdir)
>       assert result.parseoutcomes() == {"passed": 1}
E       AssertionError: assert {'passed': 2} == {'passed': 1}
E         
E         Differing items:
E         {'passed': 2} != {'passed': 1}
E         Use -v to get more diff
```

--- a/AUTHORS
+++ b/AUTHORS
@@ -320,6 +320,7 @@ Mike Hoyle (hoylemd)
 Mike Lundy
 Milan Lesnek
 Miro Hrončok
+Mulat Mekonen
 mrbean-bremen
 Nathan Goldbaum
 Nathan Rousseau

--- a/changelog/13771.contrib.rst
+++ b/changelog/13771.contrib.rst
@@ -1,0 +1,1 @@
+Skip `test_do_not_collect_symlink_siblings` on Windows environments without symlink support to avoid false negatives.

--- a/testing/test_collection.py
+++ b/testing/test_collection.py
@@ -1871,7 +1871,8 @@ def test_do_not_collect_symlink_siblings(
     """
     # Use tmp_path because it creates a symlink with the name "current" next to the directory it creates.
     symlink_path = tmp_path.parent / (tmp_path.name[:-1] + "current")
-    assert symlink_path.is_symlink() is True
+    if not symlink_path.is_symlink():  # pragma: no cover
+        pytest.skip("Symlinks not supported in this environment")
 
     # Create test file.
     tmp_path.joinpath("test_foo.py").write_text("def test(): pass", encoding="UTF-8")

